assets/js/googlesitekit/modules/datastore/sharing-settings.js

```diff
@@ -0,0 +1,301 @@
+		);
+		return {
+			payload: {
+				moduleSlug,
+				management,
+			},
+			type: SET_SHARING_MANAGEMENT,
+		};
+	},
+
+	/**
+	 * Sets the sharing settings management role of given module.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {string}   moduleSlug Module slug.
+	 * @param {string[]} roles      List of roles the module is shared with.
+	 * @return {Object} Action for SET_SHARED_ROLES.
+	 */
+	setSharedRoles( moduleSlug, roles ) {
+		invariant( moduleSlug, 'moduleSlug is required.' );
+		invariant(
+			Array.isArray( roles ) &&
+				roles.every( ( role ) => typeof role === 'string' ),
+			'roles must be an array of strings.'
+		);
+		return {
+			payload: {
+				moduleSlug,
+				roles,
+			},
+			type: SET_SHARED_ROLES,
+		};
+	},
+
+	/**
+	 * Saves sharingSettings for dashboard sharing.
+	 *
+	 * Save sharingSettings for dashboard sharing.
+	 * Update ownerID from the response for the modules in the sharingSettings state.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return {Object} Object with `{response, error}`.
+	 */
+	*saveSharingSettings() {
+		const registry = yield Data.commonActions.getRegistry();
+		// TODO: Refactor to use `getSharingSettings` selector
+		// to obtain the `sharingSettings` from the state in 4795.
+		const { sharingSettings } = registry.stores[
+			CORE_MODULES
+		].store.getState();
+
+		const {
+			response,
+			error,
+		} = yield fetchSaveSharingSettingsStore.actions.fetchSaveSharingSettings(
+			sharingSettings
+		);
+
+		// Update module owner IDs in the sharing settings modules.
+		if ( response?.newOwnerIDs ) {
+			yield baseActions.setOwnerID( response.newOwnerIDs );
+		}
+
+		return { response, error };
+	},
+
+	/**
+	 * Receives sharingSettings for dashboard sharing.
+	 * Stores sharingSettings in the datastore.
+	 *
+	 * Because this is frequently-accessed data, this is usually sourced
+	 * from a global variable (`_googlesitekitDashboardSharingData`), set by PHP
+	 * in the `before_print` callback for `googlesitekit-datastore-site`.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} sharingSettings Sharing settings for modules with `management` and `sharedRoles` properties.
+	 * @return {Object} Action for RECEIVE_GET_SHARING_SETTINGS.
+	 */
+	receiveGetSharingSettings( sharingSettings ) {
+		invariant( sharingSettings, 'sharingSettings is required.' );
+		return {
+			payload: { sharingSettings },
+			type: RECEIVE_GET_SHARING_SETTINGS,
+		};
+	},
+};
+
+const baseReducer = ( state, { type, payload } ) => {
+	switch ( type ) {
+		case SET_SHARING_MANAGEMENT: {
+			const { moduleSlug, management } = payload;
+
+			return {
+				...state,
+				sharingSettings: {
+					...state.sharingSettings,
+					[ moduleSlug ]: {
+						...state.sharingSettings[ moduleSlug ],
+						management,
+					},
+				},
+				savedSharingSettings: {
+					...state.savedSharingSettings,
+					[ moduleSlug ]: {
+						...state.savedSharingSettings[ moduleSlug ],
+						management,
+					},
+				},
```

Comment:
@hussain-t, only `sharingSettings` should be updated because `savedSharingSettings` object represents the original sharing settings and it shouldn't be changed until settings are saved. Same applies for other actions in this reducer, except `RECEIVE_GET_SHARING_SETTINGS`.